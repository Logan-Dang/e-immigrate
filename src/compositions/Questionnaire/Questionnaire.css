.Questionnaire {
}

.Questionnaire .Question {
    border: solid black 1px;
    margin: 1rem;
    padding: 1rem;
}

.Questionnaire .Question .QuestionText {
    margin-bottom: 0.5rem;
}
