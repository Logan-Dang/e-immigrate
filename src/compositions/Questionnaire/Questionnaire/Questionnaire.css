.Questionnaire {
    max-width: 600px;
    margin: auto;
    display: grid;
}

.Questionnaire button {
    margin: auto;
}
