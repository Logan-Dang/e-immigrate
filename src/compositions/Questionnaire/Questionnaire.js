import React, { useMemo } from 'react';

import './Questionnaire.css';

const Question = ({ question }) => {
    const {
        id,
        category,
        text,
        questionType,
        answerSelections,
        followUp,
        required,
    } = question;
    const answers = useMemo(() => {
        if (answerSelections) {
            return answerSelections.split(',');
        }
    }, [answerSelections]);

    const Input = () => {
        switch (questionType) {
            case 'date':
                return <input type="date" />;
            case 'radio':
                return (
                    <div>
                        {answers &&
                            answers.map((option) => (
                                <div key={option}>
                                    <input
                                        type="radio"
                                        id={option}
                                        name={`radio${id}`}
                                        // value={option}
                                    />
                                    <label htmlFor={option}>{option}</label>
                                </div>
                            ))}
                    </div>
                );
            case 'checkbox':
                return (
                    <div>
                        {answers &&
                            answers.map((option) => (
                                <div key={option}>
                                    <input
                                        type="checkbox"
                                        id={option}
                                        name={option}
                                        value={option}
                                    />
                                    <label for={option}>{option}</label>
                                </div>
                            ))}
                    </div>
                );
            case 'input':
                return <input type="text" />;
            case 'textArea':
                return <textarea rows="4" cols="50" />;
            case 'dropDown':
                return (
                    <select>
                        {answers &&
                            answers.map((option) => {
                                return <option key={option}>{option}</option>;
                            })}
                    </select>
                );
            case 'email':
                return <input type="email" id="email" name="email" />;
            case 'phoneNumber':
                return (
                    <input type="tel" pattern="[0-9]{3}-[0-9]{2}-[0-9]{3}" />
                );
            case 'zip':
                return (
                    <input type="text" id="zip" name="zip" pattern="[0-9]*" />
                );
        }
    };

    return (
        <div className="Question">
            <div className="QuestionText">
                {text}
                {required ? ' (required)' : ' (optional)'}
            </div>
            <Input className="QuestionInput" />
        </div>
    );
};

const Questionnaire = ({ questions }) => {
    return (
        <div className="Questionnaire">
            {questions.map((question) => {
                return <Question key={question.id} question={question} />;
            })}
        </div>
    );
};

export default Questionnaire;
