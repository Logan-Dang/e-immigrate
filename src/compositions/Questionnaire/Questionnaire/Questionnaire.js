import React from 'react';
import Button from '../../../components/Button/Button';
import Question from '../Question/Question';
import useMarkFieldAsTouched from '../hooks/useMarkFieldAsTouched';

import './Questionnaire.css';

const Questions = ({ filteredQuestions, bindField, questions }) => (
    <>
        {filteredQuestions.map((question) => {
            if (question.parentQuestionSlug) {
                return null;
            }

            return (
                <Question
                    key={question.id}
                    question={question}
                    className="FormElement"
                    bindField={bindField}
                    followUpQuestions={questions.filter(
                        (q) => q.parentQuestionSlug === question.slug
                    )}
                />
            );
        })}
    </>
);

const Questionnaire = ({ questions }) => {
    const filteredQuestions = questions.filter(
        (q) => q.category === 'Red Flag'
    );
    const [bindField, setAllFieldsTouched] = useMarkFieldAsTouched();

    return (
        <form
            className="Questionnaire"
            onSubmit={(event) => {
                event.preventDefault();
                const formData = new FormData(event.target);
                const data = Object.fromEntries(formData.entries());
            }}
        >
            <Questions
                filteredQuestions={filteredQuestions}
                bindField={bindField}
                questions={questions}
            />
            <Button
                label={'Submit'}
                type="submit"
                className="FormElement"
                onClick={setAllFieldsTouched}
            />
        </form>
    );
};

export default Questionnaire;
