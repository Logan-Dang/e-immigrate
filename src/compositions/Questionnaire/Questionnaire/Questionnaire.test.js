import React from 'react';
import Questionnaire from './Questionnaire';

import renderer from 'react-test-renderer';

it('App renders correctly', () => {
    const tree = renderer.create(<Questionnaire />).toJSON();
    expect(tree).toMatchSnapshot();
});
