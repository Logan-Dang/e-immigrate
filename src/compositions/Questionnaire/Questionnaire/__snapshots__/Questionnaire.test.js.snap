// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Questionnaire renders correctly 1`] = `
<div
  className="Questionnaire"
>
  <button
    className="buttonComponent"
    onClick={[Function]}
  />
</div>
`;
